components/script/script_module.rs

```diff
@@ -1870,6 +1933,15 @@ pub(crate) fn fetch_inline_module_script(
 pub(crate) type ModuleSpecifierMap = IndexMap<String, Option<ServoUrl>>;
 pub(crate) type ModuleIntegrityMap = IndexMap<ServoUrl, String>;
 
+/// <https://html.spec.whatwg.org/multipage/#specifier-resolution-record>
+#[derive(Default, Eq, Hash, JSTraceable, MallocSizeOf, PartialEq)]
+pub(crate) struct ResolvedModule {
+    base_url: String,
+    specifier: String,
+    #[no_trace]
+    specifier_url: Option<ServoUrl>,
```

Comment:
Please also document the fields, for example here with `/// <https://html.spec.whatwg.org/multipage/#specifier-resolution-record-as-url>`.